### PR TITLE
Fix non-string vars in --inventory output

### PR DESCRIPTION
**Pull request by ctrlok**
I receive a error when I try to use list in output variables: 

```
[all:vars]
panic: interface conversion: interface {} is []interface {}, not string

goroutine 1 [running]:
main.cmdInventory(0x11d35c0, 0xc42000c018, 0x11d35c0, 0xc42000c020, 0xc42000a0e0, 0x0)
	/Users/me/go/src/github.com/adammck/terraform-inventory/cli.go:77 +0x65a
main.main()
	/Users/me/go/src/github.com/adammck/terraform-inventory/main.go:100 +0x4be
```

### CODE REVIEW

**Comment by adammck** [comment]
Thanks for the patch! Would you mind adding an example to parser_test.go, so we can see how this is supposed to work? I think this tool may originally have been written before Terraform had output types other than string.

**Comment by ctrlok** [comment]
@adammck I tried to add some tests, but I'm not sure, because I can't check inventory as string, so I add tests from ansible and check equality by counting chars.

**Comment by adammck** [comment]
I'm sorry, I don't quite understand the need for counting characters. It seems like we should be able to compare expectedInventoryOutput with stdout? What's different about the output?

**Comment by ctrlok** [comment]
@adammck I do that, because 'expectedInventoryOutput' is unsorted.

**Comment by adammck** [comment]
Hmm, good point, the output isn't deterministic because we're just iterating the map and dumping it out. How about we sort the keys and iterate that, instead? That'd be more sanely testable, and much better for people who use this tool to dump the inventory and check it into a repo somewhere, too.

**Comment by ctrlok** [comment]
@adammck Sure. I didn't want to change anything before discuss :)
Also, I find a bug in my code, so I will fix it.

**Comment by ctrlok** [comment]
@adammck looks like it works :)

**Comment by adammck** [comment]
Excellent, thanks very much for doing the extra work there. This is a huge improvement.

File: cli.go [hunk 1]
```diff
@@ -73,8 +78,15 @@ func cmdInventory(stdout io.Writer, stderr io.Writer, s *state) int {
...
 			writeLn("["+group+":vars]", stdout, stderr)
-			for key, item := range grp.Vars {
-				writeLn(key+"="+item.(string), stdout, stderr)
+			vars := []string{}
+			for key, _ := range grp.Vars {
+				vars = append(vars, key)
+			}
+			sort.Strings(vars)
+			for _, key := range vars {
+				jsonItem, _ := json.Marshal(grp.Vars["key"])
```
Comment (adammck) [inline cli.go]: Isn't this supposed to be `grp.Vars[key]` ?

File: parser_test.go [hunk 2]
```diff
@@ -332,6 +332,147 @@ const expectedListOutput = `
...
+10.2.1.5
+10.20.30.40
+192.168.0.3
+50.0.0.1
+
+[all:vars]
+datacenter=null
+ids=null
+map=null
+olddatacenter=null
```
Comment (adammck) [inline parser_test.go]: Looks like the above error manifests here. These should probably match the vars at :284, in expectedListOutput.

Comment (ctrlok) [reply parser_test.go]: Wow. I'm sorry.

Comment (adammck) [reply parser_test.go]: It's no problem, I almost missed it myself.

**Comment by ctrlok** [comment]
@adammck done. Also, I tested it in ansible for sure :)

**Pull request merged**